Src/Sarif.PatternMatcher.Cli/Program.cs

```diff
@@ -53,7 +56,7 @@ internal static int Main(string[] args)
               .MapResult(
                 (AnalyzeDatabaseOptions options) => new AnalyzeDatabaseCommand().Run(options),
                 (ImportAndAnalyzeOptions options) => new ImportAndAnalyzeCommand().Run(options),
-                (AnalyzeOptions options) => new AnalyzeCommand(FileSystem).Run(options),
+                (AnalyzeOptions options) => RunAnalyzeCommand(options),
```

Comment:
>RunAnalyzeCommand [](http://example.com/codeflow?start=44&length=17)

So here's a proposal for a test pattern as people add unit tests. The condition I need to check provokes a failure at the CommandLine library level (i.e., the analyze command is never invoked).

So I've created a pattern where on actually calling into the analyze command, we retain the command instance. For this case, a null value means we failed core command-line parsing, as expected. Moving forward, this pattern will actually enable unit-testing at a much higher level, you can simply pass test args to Main and do the work. This increase confidence in testing as more product code is actually utilized. Lots of helper code could wither away with this approach.

I'm putting out this single change to experiment/refine further. i.e., not over-investing in covering existing code. As usual, for addressing technical debt, my common approach is to work on things are they churn. Having landed on a good approach we could consider going back for scrubs, if helpful.

Reply:
Something else that's pretty great about this approach is you can check/validate the exit reason.

Reply:
Yes, exactly, that's we 'poke' the command into static data, so you can check the RuntimeErrors, look at any unhandled exception that was raised, etc.